int32 dummy
---
int32[] points